### DISCUSSION TITLE
Use composite keys for Ids

### DISCUSSION THREAD

**Discussion by youcefTab** [refinedev/refine#2016] [Q&A] [answered]
Hey guys, thanks for the amazing work.

In my back-end i have a resource that does not have an id, the is used to map the resource is a composite key ( 3 columns used as a primary key ), using the base key of refine to delete for example, i can only provide an id as a string, so it's a limitation for me.

I was wandering if there is no solutions already available,  if it's not the case i think it would be a good update to do.

**Comment by omeraplak** [comment, marked as answer]
Hey @youcefTab ,
Thank you for your kind words! What you can do in this situation is, You will pass information from `data`, `form`, `table` hooks to dataProvider using `metaData`. 

We made an example in [`real-world-example`](https://github.com/gothinkster/realworld),

Usage,
https://github.com/pankod/refine/blob/next/examples/real-world-example/src/pages/article/index.tsx#L101

Data Provider,
https://github.com/pankod/refine/blob/next/examples/real-world-example/src/dataProvider.ts#L92

Example,
https://refine.dev/docs/examples/real-world-refine-example/

I hope this will be useful for you 🎁
